### ISSUE TITLE
UI features suggestions

### ISSUE THREAD

**Issue by xLn2** [NicoAiko/mikazuki#218]
### **1. Dynamic cursor on scale rendering**

**Is your feature request related to a problem? Please describe.**

**1.  Mouse over render score**
2. 1 and 2 is same bullet
Currently with on click action scoring is printed on. So I have suggestion that > cursor on scale's point will render existing score at Score `(0)`. 

**3. Minus and Plus (1) Episode**

The thing is there is `plus (+)` button in [/application/#/aniList/watching](https://www.mikazuki.moe/application/#/aniList/watching) but not in page of anime (i.e: /application/#/detail-view/249) and other pages i.e [/application/#/aniList/paused](https://www.mikazuki.moe/application/#/aniList/paused) 
I sometimes frustrated about where was minus button and go to anime page and there is stilll no minus button just keyboard friendly textbox so I did some graphical representation about what I imagined about.

![mikazuki test](https://user-images.githubusercontent.com/3073922/101991011-ef38e900-3cba-11eb-8843-bb2ad32dd16b.png)

**4. Anime progress on bar**
As above image described there is bar below "Progress" that represent current watched episodes and how many episodes left.
The image below just for represent on mouse on specific area shows value of current episodes and left episodes. (taken from anilist)
![image](https://user-images.githubusercontent.com/3073922/101990116-e265c680-3cb5-11eb-9baf-1746a12b1f6f.png)

### **2. Anime page improvement** 

1. At 1-4 mentioned these settings can be integrated through anime list pages at the bottom of anime picture (watching etc.).
And mean average score can be visible on top-left corner of the anime pic.

![mikazuki test 2](https://user-images.githubusercontent.com/3073922/101988747-bb56c700-3cac-11eb-8131-cf5cdc917f2e.png) rank i.e ![image](https://user-images.githubusercontent.com/3073922/101988839-58196480-3cad-11eb-935b-a7f8e82924d9.png)[Example](https://www.turkanime.net/anime/100-man-no-inochi-no-ue-ni-ore-wa-tatteiru)

(in anime page) Description part is very far from page (at the bottom) it may rise 👍

### **3. Drag Pointer on 100 scale system**

There is currently available drag pointer system in Mikazuki but not icluded in anime page (100scale)
![mikazuki test 3](https://user-images.githubusercontent.com/3073922/101988753-c6a9f280-3cac-11eb-8e2e-28ca35fbab4b.png)

![image](https://user-images.githubusercontent.com/3073922/101990573-ab44e480-3cb8-11eb-9fe0-4ff505db7904.png)

### **4. Minus Episode [Global]**

I prepared a template that where should `minus (-)` been. As template shows there is option that `at the beginning` of bar or next to the `plus (+)`
![mikazuki test 4](https://user-images.githubusercontent.com/3073922/101988757-d1fd1e00-3cac-11eb-85fe-8f0e43f84b18.png)

_**These are my suggestions, I am sorry if I used wrong or offensive language. I want to see what is your opinions about these suggestion. Lets discuss! 🎉**_

**Comment by NicoAiko** [comment]
First of all, thanks for these suggestions.
I'll do some comments on these points now.

# 1. Dynamic cursor on scale rendering

## 1./2. Scoring
I'd most likely apply the slider way to all scoring elements to keep it the same way.
The Details page didn't get any real attention when I rebuild the list view, so that's still something that's planned.

## 3. Minus and Plus button on episode inputs
I wasn't implementing a minus button on purpose, because I thought, at least on the regular list view, nobody would really need a minus button. I can completely understand the desire to have one though, so that's a thing I can of course implement.

As far as the buttons in the input field (or at least in the details page) is concerned, I'd like to discuss in more detail about how to properly implement it, as I'm not that fond of having buttons inside an input field.
There should be a design change for that.
(Also sorry for frustrating you concerning that... xD)

As to why there are no plus buttons on any lists except repeat and current is actually ... logic in my opinion.

Let's take for example an entry that you have on your Completed list. That entry's episode progress should already be on 100%. Furthermore, if you're watching it (again), you'd most likely have it put on your Rewatching list, where you can increase your progress again.

The same can be applied to an entry you dropped, paused or plan to watch.

## 4. Anime progress on bar
Sure, so basically the same as on the list view but inside the details view.

# 2. Anime page improvement

## 1. Episode bars
With this one, I don't really understand what you mean. Mikazuki already has bars that represent how many episodes you've watched and (if still airing) how many episodes there still are for you to watch until you are at the latest released one.
Could you elaborate in more detail what you're meaning here?

## 2. Scorings on the image
This I'm actually against. Let's say we take your idea as a base. How would the user know what scoring is theirs? There would have to be an icon or some sort of indicator that distincts the two.
It depends on the page you want to show it. In the regular list view for your anime? I'd be against this.
On the Search view or Season Preview? Sure why not. The context is important in my opinion.

## 2.5 Description
Yes, I agree that the description should be put higher. I'll have to rearrange and change Designs there though, because there are long and short descriptions.

## 3. Score pointer
See Scoring (1./2.) above.

## 4. Minus button
Basically also Minus and Plus button on episode inputs (1.3.) above.
If it's wished though, I can implement it.

I hope I could answer some stuff :)

**Comment by nattadasu** [comment]
FYI, for question number 1.3, and I'll take a *peek* from another stable app:
In some tracking apps, like MyAniList for iOS, Taiga for Windows, and Trackma for Linux/UNIX, or even that doesn't relate to anime: TV Time, they do provide users on Increase/Decrease episode button/gesture on their application for PTW/On-hold/dropped list (ofc there's no button/gesture on completed list).

In execution, when user tap/click/do "increase episode" button/gesture on *plan to watch* or *on-hold*, it'd be automatically move that show to _current/watching_ list.

It would be nice if Mikazuki can do this, r8? ~~Or did I even miss this?~~ 🤔

**Comment by NicoAiko** [comment]
> FYI, for question number 1.3, and I'll take a _peek_ from another stable app:
> In some tracking apps, like MyAniList for iOS, Taiga for Windows, and Trackma for Linux/UNIX, or even that doesn't relate to anime: TV Time, they do provide users on Increase/Decrease episode button/gesture on their application for PTW/On-hold/dropped list (ofc there's no button/gesture on completed list).
> 
> In execution, when user tap/click/do "increase episode" button/gesture on _plan to watch_ or _on-hold_, it'd be automatically move that show to _current/watching_ list.
> 
> It would be nice if Mikazuki can do this, r8? ~Or did I even miss this?~ 🤔

That's also a possibility. Nice idea.

**Comment by xLn2** [comment]
> In execution, when user tap/click/do "increase episode" button/gesture on plan to watch or on-hold, it'd be automatically move that show to current/watching list.

I have suggestion about it. At these scerenios there is another information box can appear to ask "**Do you want to change media to watching list?**".

**Comment by xLn2** [comment]
>With this one, I don't really understand what you mean. Mikazuki already has bars that represent how many episodes you've watched and (if still airing) how many episodes there still are for you to watch until you are at the latest released one.
Could you elaborate in more detail what you're meaning here?

My view is wholeness about design. When I switch from list to details page it should not be different so much I mean I can conduct most of the things without changing page. This can be thin and shows with 'overflow:visible'

>2-1. Episode bars
With this one, I don't really understand what you mean. Mikazuki already has bars that represent how many episodes you've watched and (if still airing) how many episodes there still are for you to watch until you are at the latest released one.
Could you elaborate in more detail what you're meaning here?

Actually its same 1 above so the position of bar can be placed in bottom of anime cover or desired location. I can link one exapmle mikazuki mobile app bar is good example. 

>2-2. Scorings on the image
It depends on the page you want to show it. In the regular list view for your anime? I'd be against this.
On the Search view or Season Preview? Sure why not. The context is important in my opinion.

Yes it may lead confusion. Might as you mention season preview good place for it. And maybe show (scale/100) for not listed animes global setting. Example in home page one of my friend added an anime so this is not my list in this case can be evaluated for project.

**Comment by InFerNoBeasTz** [comment]
Would be awesome if you add sort by popular, trending: daily, weekly and monthly.

**Comment by NicoAiko** [comment]
> Would be awesome if you add sort by popular, trending: daily, weekly and monthly.

I'll look into it!

**Comment by InFerNoBeasTz** [comment]
> > Would be awesome if you add sort by popular, trending: daily, weekly and monthly.
> 
> I'll look into it!

Thanks
